File: backend/app/schemas/static_layer.py
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field, HttpUrl


class StaticFeature(BaseModel):
    external_id: str = Field(min_length=1)
    name: str = Field(min_length=1)
    source: Literal["gem", "phivolcs"]
    source_url: HttpUrl
    license_name: str = Field(min_length=1)
    dataset_version: str = Field(min_length=1)
    geometry: dict[str, Any]
    source_properties: dict[str, Any] = Field(default_factory=dict)


class StaticLayer(StaticFeature):
    id: str
    imported_at: datetime

File: web/src/types/staticLayer.ts
import type { LineString, MultiLineString, Polygon, MultiPolygon } from 'geojson';

export type StaticLayer = {
  id: string;
  external_id: string;
  name: string;
  source: 'gem' | 'phivolcs';
  source_url: string;
  license_name: string;
  dataset_version: string;
  imported_at: string;
  source_properties: Record<string, unknown>;
  geometry: LineString | MultiLineString | Polygon | MultiPolygon;
};
export type StaticLayerKind = 'faults' | 'volcano-zones';
